packages/dma-contracts/test/e2e/strategies/ajna/close.test.ts

```diff
@@ -0,0 +1,306 @@
+  const supportedPositions = getSupportedAjnaPositions(networkFork)
+  let env: EnvWithAjnaPositions
+  const fixture = envWithAjnaPositions({
+    network: networkFork,
+    systemConfigPath: `test/${networkFork}.conf.ts`,
+    configExtensionPaths: [`test/uSwap.conf.ts`],
+    hideLogging: true,
+  })
+  before(async function () {
+    env = await loadFixture(fixture)
+    if (!env) throw new Error('Env not setup')
+  })
+
+  type Token = {
+    symbol: string
+    precision: number
+    address: Address
+  }
+
+  describe('Close Positions using Close to Quote', function () {
+    supportedPositions.forEach(({ name: variant }) => {
+      let position: AjnaPosition
+      let debtToken: Token
+      let collateralToken: Token
+      let positionDetails: AjnaPositionDetails
+      let act: Unbox<ReturnType<typeof closePositionHelper>>
+
+      before(async function () {
+        const { positions, ajnaSystem } = env
+        positionDetails = positions[variant]
+        if (!positionDetails) {
+          throw new Error('Position not found')
+        }
+        position = await views.ajna.getPosition(
+          {
+            proxyAddress: positionDetails.proxy,
+            poolAddress: positionDetails.pool.poolAddress,
+            collateralPrice: positionDetails.__collateralPrice,
+            quotePrice: positionDetails.__quotePrice,
+          },
+          {
+            poolInfoAddress: ajnaSystem.poolInfo.address,
+            provider: env.config.provider,
+            getPoolData: env.dependencies.getPoolData,
+          },
+        )
+        debtToken = positionDetails.debtToken
+        collateralToken = positionDetails.collateralToken
+
+        act = await closePositionHelper({
+          env,
+          positionDetails,
+          position,
+          shouldCloseToCollateral: false,
+        })
+      })
+
+      it(`Should have closed ${variant} position`, async () => {
+        expect(act.closeTxStatus).to.be.true
+      })
+      it(`Should have paid back all debt for ${variant}`, async () => {
+        expect.toBe(act.closedPosition.debtAmount.toFixed(0), 'lte', ONE)
+      })
+      it(`Should have withdrawn all collateral for ${variant}`, async () => {
+        expect.toBe(act.closedPosition.collateralAmount.toFixed(0), 'lte', ONE)
+      })
+      it(`Should not have anything left on the proxy for ${variant}`, async () => {
+        const proxyDebtBalance = await balanceOf(debtToken.address, positionDetails.proxy, {
+          config: env.config,
+          isFormatted: true,
+        })
+        const proxyCollateralBalance = await balanceOf(
+          collateralToken.address,
+          positionDetails.proxy,
+          {
+            config: env.config,
+            isFormatted: true,
+          },
+        )
+
+        expect.toBeEqual(proxyDebtBalance, ZERO)
+        expect.toBeEqual(proxyCollateralBalance, ZERO)
+      })
+      it(`Should have collected a fee for ${variant}`, async () => {
```

Comment:
We should also test whether the user got the collateral afterwards.

Reply:
I'll implement one. Thanks dude 💯